daemon/k8s_watcher.go

```diff
@@ -99,12 +99,14 @@ func (m *k8sAPIGroupsUsed) removeAPI(api string) {
 	delete(m.apis, api)
 }
 
-func (m *k8sAPIGroupsUsed) Range(f func(key string, value bool) bool) {
-	for k, v := range m.apis {
-		if !f(k, v) {
-			return
-		}
+func (m *k8sAPIGroupsUsed) getGroups() []string {
+	m.RLock()
+	defer m.RUnlock()
+	groups := make([]string, len(m.apis))
+	for k := range m.apis {
+		groups = append(groups, k)
```

Comment:
```
i := 0
i++
groups[i] = k
```
otherwise `k8sAPIGroupsUsed` will be something like `["", "", "", "foo", "bar", "baz"]`

Reply:
oops. You are right. I should have set the slice to len==0. Fixed.